src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,18 +727,25 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
+    if (storyDelta === 0) {
+      return;
+    }
     if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
+      !this.circularWrapping_() &&
+      this.isIndexOutofBounds_(this.currentIdx_ + storyDelta)
     ) {
       throw new Error('Out of Story range.');
     }
-    if (storyDelta === 0) {
-      return;
+    let currentStory;
+    const newIdx = this.currentIdx_ + storyDelta;
+    if (!this.isIndexOutofBounds_(newIdx)) {
+      currentStory = this.stories_[newIdx];
+    } else {
+      currentStory =
+        storyDelta > 0
+          ? this.stories_[newIdx % this.stories_.length]
+          : this.stories_[(newIdx % this.stories_.length + this.stories_.length) % this.stories_.length];
```

Comment:
Sorry, just thought of a way to simplify this more. Since we're already using `%` for out of bounds indices, we don't need the `if(!this.isIndexOutOfBounds_(newIdx)` check.

```suggestion
    
  currentStory =
    storyDelta > 0
       ? this.stories_[newIdx % this.stories_.length]
       : this.stories_[(newIdx % this.stories_.length + this.stories_.length) % this.stories_.length];
```